docs/architecture/grpc-for-wcf-developers/wcf-endpoints-grpc-methods.md

```diff
@@ -0,0 +1,47 @@
+---
+title: WCF endpoints and gRPC methods - gRPC for WCF Developers
+description: Comparison of WCF endpoints declared with ServiceContract and OperationContract attributes, and gRPC methods declared in Protobuf
+author: markrendle
+ms.date: 09/02/2019
+---
+
+# WCF endpoints and gRPC methods
+
+In WCF, when you're writing your application code, you use one of the following methods:
+
+- You write the application code in a class and decorate methods with the [OperationContract](xref:System.ServiceModel.OperationContractAttribute) attribute.
+- You declare an interface for the service and add [OperationContract](xref:System.ServiceModel.OperationContractAttribute) attributes to the interface.
+
+For example, the WCF equivalent of the `greet.proto` Greeter service might be written as follows:
+
+```csharp
+[ServiceContract]
+public interface IGreeterService
+{
+    [OperationContract]
+    string SayHello(string name);
+}
+```
+
+Chapter 3 showed that Protobuf message definitions are used to generate data classes. Service and method declarations are used to generate base classes that you inherit from to implement the service. You just declare the methods to be implemented in the `.proto` file, and the compiler generates a base class with virtual methods you must override.
+
+## OperationContract properties
+
+The [OperationContract](xref:System.ServiceModel.OperationContractAttribute) attribute has properties to control or refine how it works. gRPC methods don’t offer this type of control. The following table sets out those `OperationContract` properties and how the functionality they specify is (or isn’t) dealt with in gRPC:
+
+| `OperationContract` property | gRPC                                             |
+| ---------------------------- | ------------------------------------------------ |
+| <xref:System.ServiceModel.OperationContractAttribute.Action>             | SOAP-related, no meaning in gRPC. |
+| <xref:System.ServiceModel.OperationContractAttribute.AsyncPattern>       | All gRPC service methods return `Task` objects. |
+| <xref:System.ServiceModel.OperationContractAttribute.HasProtectionLevel> | No message encryption; network encryption handled at the transport layer (TLS over HTTP/2). |
+| <xref:System.ServiceModel.OperationContractAttribute.IsInitiating>       | See [gRPC streaming](rpc-types.md#grpc-streaming). |
+| <xref:System.ServiceModel.OperationContractAttribute.IsOneWay>           | One-way gRPC methods return `Empty` results or use client streaming. |
+| <xref:System.ServiceModel.OperationContractAttribute.IsTerminating>      | See [gRPC streaming](rpc-types.md#grpc-streaming). |
```

Comment:
I didn't see any reference to this in rpc-types.md. When an operation has this property set, it causes the server to initiate closing the session/connection (depending on transport) after it has sent the response to this operation. Can also be set on a callback contract and when the server calls that callback operation, the client will close the channel after the call completes.